tests/test_optimize.py

```diff
@@ -19,6 +19,7 @@
 
 import numpy as onp
 import pytest
+from pytest_mock import mocker
```

Comment:
No need to import `mocker` here; `mocker` is a fixture, so pytest will automatically perform 'dependency injection' so that Python can find it in required test functions. Not a very pythonic system!
```suggestion
from pytest_mock import mocker
```